controllers/space/space_controller_test.go

```diff
@@ -1180,6 +1211,100 @@ func TestUpdateSpaceTier(t *testing.T) {
 		})
 	})
 
+	t.Run("update needed due to feature toggle annotation change in space", func(t *testing.T) {
+		type testRun struct {
+			name             string
+			oldSpaceFeatures string
+			newSpaceFeatures string
+		}
+
+		tests := []testRun{
+			{
+				name:             "feature enabled",
+				oldSpaceFeatures: "",
+				newSpaceFeatures: "feature-1,feature-2",
+			},
+			{
+				name:             "feature disabled",
+				oldSpaceFeatures: "feature-1,feature-2",
+				newSpaceFeatures: "",
+			},
+			{
+				name:             "feature updated",
+				oldSpaceFeatures: "feature-1,feature-2",
+				newSpaceFeatures: "feature-3",
+			},
+			{
+				name:             "no updates",
+				oldSpaceFeatures: "feature-1,feature-2",
+				newSpaceFeatures: "feature-1, feature-2",
+			},
+			{
+				name:             "features still are not present",
+				oldSpaceFeatures: "",
+				newSpaceFeatures: "",
+			},
+		}
+		for _, testRun := range tests {
+			t.Run(testRun.name, func(t *testing.T) {
+				// given
+
+				// Create the initial Space and NSTemplateSet with features if needed which we will to attempt to update later
+				spaceOptions := []spacetest.Option{spacetest.WithSpecTargetCluster("member-1")}
+				if testRun.oldSpaceFeatures != "" {
+					spaceOptions = append(spaceOptions, spacetest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.oldSpaceFeatures))
+				}
+
+				s := spacetest.NewSpace(test.HostOperatorNs, "oddity", spaceOptions...)
+				hostClient := test.NewFakeClient(t, s, base1nsTier)
+				member1 := NewMemberClusterWithTenantRole(t, "member-1", corev1.ConditionTrue)
+				ctrl := newReconciler(hostClient, member1)
+
+				_, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+				require.NoError(t, err)
+				s = spacetest.AssertThatSpace(t, test.HostOperatorNs, s.Name, hostClient).
+					Exists().Get()
+				nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, s.Name, member1.Client).
+					Exists().Get()
+				nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
+					{
+						Type:   toolchainv1alpha1.ConditionReady,
+						Status: corev1.ConditionTrue,
+						Reason: toolchainv1alpha1.NSTemplateSetProvisionedReason,
+					},
+				}
+				err = member1.Client.Update(context.TODO(), nsTmplSet)
+				require.NoError(t, err)
+
+				// when
+				// update the space annotation
+				if testRun.newSpaceFeatures != "" {
+					s.Annotations = make(map[string]string)
+					s.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey] = testRun.newSpaceFeatures
+				} else {
+					delete(s.Annotations, toolchainv1alpha1.FeatureToggleNameAnnotationKey)
+				}
+				err = hostClient.Update(context.TODO(), s)
+				require.NoError(t, err)
+
+				_, err = ctrl.Reconcile(context.TODO(), requestFor(s))
+				require.NoError(t, err)
```

Comment:
Calling the reconcile twice as part of one unit-test is not ideal (for the sake of isolation, and also for the sake of consistency). 
You can either split it into two tests (parent and sub-test), but this would mean that the first test duplicates one from the top. 
Or modify the setup so it defines the outdated NSTempateSet explicitly - it's easy to understand the setup and the purpose of the test, and it's consistent with the rest of the unit tests:
```suggestion
				// given
				spaceOptions := []spacetest.Option{spacetest.WithSpecTargetCluster("member-1")}
				if testRun.newSpaceFeatures != "" {
					spaceOptions = append(spaceOptions, spacetest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.newSpaceFeatures))
				}

				s := spacetest.NewSpace(test.HostOperatorNs, "oddity", spaceOptions...)

				nsTemplateSetOptions := []nstemplatetsettest.Option{nstemplatetsettest.WithReadyCondition(), nstemplatetsettest.WithReferencesFor(base1nsTier)}
				if testRun.oldSpaceFeatures != "" {
					nsTemplateSetOptions = append(nsTemplateSetOptions, nstemplatetsettest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.oldSpaceFeatures))
				}
				nsTemplateSet := nstemplatetsettest.NewNSTemplateSet("oddity", nsTemplateSetOptions...)
				hostClient := test.NewFakeClient(t, s, base1nsTier)
				member1Client := test.NewFakeClient(t, nsTemplateSet)
				member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
				ctrl := newReconciler(hostClient, member1)

				// when
				_, err = ctrl.Reconcile(context.TODO(), requestFor(s))
				require.NoError(t, err)
```

Reply:
you can also add a check that the NSTemplateSet is not updated when it's not supposed to be:
```go
				if testRun.oldSpaceFeatures == testRun.newSpaceFeatures {
					member1Client.MockUpdate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.UpdateOption) error {
						return fmt.Errorf("shouldn't be called")
					}
				}
```

Reply:
so the whole proposed code is:
```suggestion
				// given
				spaceOptions := []spacetest.Option{spacetest.WithSpecTargetCluster("member-1")}
				if testRun.newSpaceFeatures != "" {
					spaceOptions = append(spaceOptions, spacetest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.newSpaceFeatures))
				}

				s := spacetest.NewSpace(test.HostOperatorNs, "oddity", spaceOptions...)

				nsTemplateSetOptions := []nstemplatetsettest.Option{nstemplatetsettest.WithReadyCondition(), nstemplatetsettest.WithReferencesFor(base1nsTier)}
				if testRun.oldSpaceFeatures != "" {
					nsTemplateSetOptions = append(nsTemplateSetOptions, nstemplatetsettest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.oldSpaceFeatures))
				}
				nsTemplateSet := nstemplatetsettest.NewNSTemplateSet("oddity", nsTemplateSetOptions...)
				hostClient := test.NewFakeClient(t, s, base1nsTier)
				member1Client := test.NewFakeClient(t, nsTemplateSet)
				if testRun.oldSpaceFeatures == testRun.newSpaceFeatures {
					member1Client.MockUpdate = func(ctx context.Context, obj runtimeclient.Object, opts ...runtimeclient.UpdateOption) error {
						return fmt.Errorf("shouldn't be called")
					}
				}
				member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
				ctrl := newReconciler(hostClient, member1)
```

Reply:
Yes, I was using the first reconcile just as part of the setup. Not the test itself. My comment could be clearer I guess.
Anyway thanks for the suggestion. It's a bit clearer what the tests is doing here I think.